modules/integration/tests-integration/lazy-loading-test/src/test/java/org/wso2/appserver/integration/lazy/loading/LazyLoadingBaseTest.java

```diff
@@ -0,0 +1,506 @@
+                    xmlStreamException.getMessage();
+            log.error(customErrorMessage);
+            throw new LazyLoadingTestException(customErrorMessage, xmlStreamException);
+
+        } catch (LoginAuthenticationExceptionException loginAuthenticationExceptionException) {
+            String customErrorMessage = "LoginAuthenticationExceptionException exception  when login as tenant admin." +
+                    loginAuthenticationExceptionException.getMessage();
+            log.error(customErrorMessage);
+            throw new LazyLoadingTestException(customErrorMessage, loginAuthenticationExceptionException);
+
+        } catch (URISyntaxException uriSyntaxException) {
+            String customErrorMessage = "URISyntaxException exception  when login as tenant admin." +
+                    uriSyntaxException.getMessage();
+            log.error(customErrorMessage);
+            throw new LazyLoadingTestException(customErrorMessage, uriSyntaxException);
+
+        }
+
+
+    }
+
+    /**
+     * check the tenant is unloading functionality when tenant is idle more than configured tenant idle time.
+     * This method will wait additional MAX_THRESHOLD_TIME milliseconds to  system to unload the tenant. it will log
+     * total idle time taken to unload.
+     *
+     * @param tenantDomain Tenant domain that need to check for unloading.
+     * @return true if tenant is unload after tenant idle time + additional MAX_THRESHOLD_TIME milliseconds if not return
+     * false
+     * @throws LazyLoadingTestException
+     */
+    protected boolean checkTenantAutoUnloading(String tenantDomain) throws LazyLoadingTestException {
+        boolean isTenantUnloaded = false;
+        log.info("Sleeping  for " + TENANT_IDLE_TIME + " milliseconds (Tenant idle tome).");
+
+        try {
+            Thread.sleep(TENANT_IDLE_TIME);
+        } catch (InterruptedException interruptedException) {
+            String customErrorMessage = "InterruptedException occurs when sleeping for TENANT_IDLE_TIME" +
+                    interruptedException.getMessage();
+            log.error(customErrorMessage);
+            throw new LazyLoadingTestException(customErrorMessage, interruptedException);
+        }
+        long totalSleepTime = 0;
+        long startTime = System.currentTimeMillis();
+        while (System.currentTimeMillis() - startTime < MAX_THRESHOLD_TIME) {
+
+            isTenantUnloaded = !getTenantStatus(tenantDomain).isTenantContextLoaded();
+            totalSleepTime = System.currentTimeMillis() - startTime + TENANT_IDLE_TIME;
+            if (isTenantUnloaded) {
+                log.info("Tenant " + tenantDomain + " is unloaded in " + totalSleepTime + "milliseconds. Tenant idle " +
+                        "time is :" + TENANT_IDLE_TIME + "milliseconds.");
+                break;
+            } else {
+                try {
+                    Thread.sleep(1000);
```

Comment:
yes, we need sleep time,
Maximum threshold time is 2 minutes.  So if we half the loop sleep time, it will double the number of server calls to  get the updated information. With comparing to the total time that loop need to sleep, by reducing from 1 minutes to half a minutes will not do much effect.